html_css/v2/main-course/selectors.md

```diff
@@ -0,0 +1,264 @@
+
+`:root` is generally the place where you will place your 'global' CSS rules that you want available everywhere - such as your custom properties and CSS variables, or rules such as `box-sizing: border-box;`.
+
+[`:first-child`](https://css-tricks.com/almanac/selectors/f/first-child/) and [`:last-child`](https://css-tricks.com/almanac/selectors/l/last-child/) will match elements that or the first or last sibling.
+
+Similarly [`:empty`](https://css-tricks.com/almanac/selectors/e/empty/) and [`:only-child`](https://css-tricks.com/almanac/selectors/o/only-child/) will match elements that have no children at all, or just one.
+
+For a more dynamic approach we can use [`:nth-child`](https://css-tricks.com/almanac/selectors/n/nth-child/). This is a flexible pseudo-class with a few different uses.
+
+~~~css
+  .myList:nth-child(5) {/* Selects the 5th child of myList */}
+
+  .myList:nth-child(3n) { /* Selects every 3rd child of myList */}
+
+  .myList:nth-child(3n + 3) { /* Selects every 3rd child of myList, beginning with the 3rd */}
+
+  .myList:nth-child(even) {/* Selects every even child of myList */}
+~~~
+
+### Pseudo-elements
+While pseudo-classes give us an alternative way to interact with our HTML elements based on their state or structure, pseudo-elements are more abstract. They allow us to affect parts of our HTML that aren't elements at all. These special elements share the same specificity as regular elements (0, 0, 0, 1). There are a number of useful pseudo-elements that can be utilized in any number of creative ways.
+
+[`::marker`](https://css-tricks.com/almanac/selectors/m/marker/) allows you to customize the styling of your `<li>` elements' bullets or numbers.
+
+[`::first-letter`](https://css-tricks.com/almanac/selectors/f/first-letter/) and [`::first-line`](https://css-tricks.com/almanac/selectors/f/first-line/) allow you to (you guessed it!) give special styling to the first letter or line of some text.
+
+[`::before` and `::after`](https://css-tricks.com/almanac/selectors/a/after-and-before/) allow us to add extra elements onto the page with CSS, instead of HTML. Using it to decorate text in various ways is one common use case.
+
+~~~html
+<style>
+  .emojify::before {
+    content: '😎 🥸 🤓';
+}
+
+  .emojify::after {
+    content: '🤓 🥸 😎';
+}
+</style>
+
+<body>
+  <div> Let's <span class="emojify">emojify</span>this span!</div>
+</body>
+~~~
+
+Using these pseudo-elements this way would give us this result:
+
+Let's 😎 🥸 🤓 emojify 🤓 🥸 😎 this span!
+
+[`::selection`](https://css-tricks.com/almanac/selectors/s/selection/) allows you to change the highlighting when a user selects text on the page.
+
+ There are lots more! Have a quick browse through the [pseudo-element docs](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements) to see a fuller list of what's possible.
```

Comment:
In my opinion "fuller" is one of those words in the English language which sounds strange phonetically, so could be replaced with:
"...to see a more substantial list of what's possible."

Reply:
changed to 'complete' which is I think more concise and more accurate